services/minespace-web/src/components/Forms/projects/projectSummary/ProjectSummaryFileUpload.tsx

```diff
@@ -0,0 +1,318 @@
+import React, { FC, useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { Field, WrappedFieldProps } from "redux-form";
+import { useSelector } from "react-redux";
+import {
+  NEW_VERSION_PROJECT_SUMMARY_DOCUMENTS,
+  PROJECT_SUMMARY_DOCUMENTS,
+} from "@common/constants/API";
+import FileUpload from "@/components/common/FileUpload";
+import { Alert, Divider, Modal, Popconfirm, Table, Typography } from "antd";
+import { getUserInfo } from "@/selectors/authenticationSelectors";
+import { FilePondFile } from "filepond";
+import { IDocument } from "@mds/common";
+import { HttpRequest, HttpResponse } from "tus-js-client";
+
+const propTypes = {
+  onFileLoad: PropTypes.func.isRequired,
+  onRemoveFile: PropTypes.func.isRequired,
+  acceptedFileTypesMap: PropTypes.objectOf(PropTypes.string).isRequired,
+  params: PropTypes.objectOf(PropTypes.string).isRequired,
+};
+
+const notificationDisabledStatusCodes = [409]; // Define the notification disabled status codes
+
+interface ProjectSummaryFileUploadProps {
+  onFileLoad: (fileName: string, document_guid: string, version_guid) => void;
+  onRemoveFile: () => void;
+  acceptedFileTypesMap: object;
+  params: any;
+  documents: IDocument[];
+}
+
+export const ProjectSummaryFileUpload: FC<WrappedFieldProps & ProjectSummaryFileUploadProps> = (
+  props
+) => {
+  const [archivedFileModalMessage, setArchivedFileModalMessage] = useState("");
+  const [isArchivedFileModalVisible, setIsArchivedFileModalVisible] = useState(false);
+  const [replaceableFileModalMessage, setReplaceableFileModalMessage] = useState("");
+  const [isReplaceableFileModalVisible, setReplaceableFileModalVisible] = useState(false);
+  const [fileName, setFileName] = useState("");
+  const [shouldReplaceFile, setShouldReplaceFile] = useState(false);
+  const [replaceFileUploadUrl, setReplaceFileUploadUrl] = useState<undefined | string>();
+  const [mineDocumentGuid, setMineDocumentGuid] = useState(null);
+  const [mineGuid, setMineGuid] = useState(null);
+  const [fileDetails, setFileDetails] = useState(null);
+  const [shouldAbortUpload, setShouldAbortUpload] = useState(false);
+  const [handleModalClose, setHandleModalClose] = useState<(() => void) | null>(null);
+  const [handleModalSubmit, setHandleModalSubmit] = useState<(() => void) | null>(null);
+  const [existingFiles, setExistingFiles] = useState([]);
+  const [version, setVersion] = useState();
+  const [uploadUrl, setUploadUrl] = useState(PROJECT_SUMMARY_DOCUMENTS(props.params));
+
+  const userInfo = useSelector(getUserInfo);
+
+  const handleCloseModal = () => {
+    handleModalClose && handleModalClose();
```

Comment:
@taraepp these are the lines that eslint was complaining about.

Reply:
Silly eslint. This looks perfectly fine to me.

I do get this message when the hook runs, I wonder if it's related: `Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .`